File: Medusa/MedusaCore/Core/Coder/CoderInfo.cpp
// Copyright (c) 2015 fjz13. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.
#include "MedusaCorePreCompiled.h"
//SIREN_BODY_INCLUDE_BEGIN
#include "CoderInfo.h"
//SIREN_BODY_INCLUDE_END

MEDUSA_BEGIN;

CoderInfo::CoderInfo()
{
//SIREN_BODY_CONSTRUCT_BEGIN
	mType = (CoderType)0;
//SIREN_BODY_CONSTRUCT_END
}

CoderInfo::~CoderInfo()
{
//SIREN_BODY_DESTRUCT_BEGIN
//SIREN_BODY_DESTRUCT_END
}

//SIREN_BODY_METADATA_BEGIN
SIREN_METADATA(CoderInfo, 9);
SIREN_PROPERTY_METADATA(0, CoderInfo, Type, 4, (CoderType)0);
SIREN_PROPERTY_METADATA_STRUCT(1, CoderInfo, Data, 4);
//SIREN_BODY_METADATA_END

MEDUSA_END;

File: Deprerated/MedusaProto/Core/CoderInfo.cs
// Copyright (c) 2015 fjz13. All rights reserved.
// Use of this source code is governed by a MIT-style
// license that can be found in the LICENSE file.
using System.Collections.Generic;
using Siren;
using Siren.Attribute;

namespace Medusa.Core
{
    [SirenClass(typeof(MedusaCoreTemplate), "Core/Coder", SirenGenerateMode.Generate | SirenGenerateMode.SuppressCompare)]
    public class CoderInfo
    {
        [SirenProperty]
        public CoderType Type { get; set; }
        [SirenProperty(SirenPropertyModifier.Optional)]
        public byte[] Data { get; set; }
    }


}
